mmv1/third_party/terraform/services/apphub/data_source_apphub_discovered_service.go

```diff
@@ -0,0 +1,185 @@
+package apphub
+
+import (
+    "fmt"
+    "time"
+
+    "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+    "github.com/hashicorp/terraform-provider-google/google/tpgresource"
+    transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceApphubDiscoveredService() *schema.Resource {
```

Comment:
I believe our precedence for data sources returning multiple resource is to pluralize them (ie. `DataSourceApphubDiscoveredServices`). See https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/dns/data_source_dns_managed_zones.go.

Reply:
We are returning a single data source instance only. We are using list instead of map because map is not supported by recent terraform version (https://github.com/hashicorp/terraform-plugin-sdk/issues/616)